core/api.go

```diff
@@ -23,20 +23,31 @@ func Version() (version string) {
 }
 
 // NewProduer create a new producer with config
-func NewProduer(config *ProducerConfig) Producer {
+func NewProduer(config *ProducerConfig) (Producer, error) {
 	return newDefaultProducer(config)
 }
 
 // ProducerConfig define a producer
 type ProducerConfig struct {
-	GroupID     string
-	NameServer  string
-	Credentials *SessionCredentials
+	GroupID          string
+	NameServer       string
+	NameServerDomain string
+	GroupName        string
+	InstanceName     string
+	Credentials      *SessionCredentials
+
```

Comment:
I think a base interface can be extracted, such as clientConfig

Reply:
fixed